### PR TITLE
Fixed incorrect behaviour of headless_renderer depending on image dimensions

### LINKED ISSUE #13384: Skewed output when changing resolution of new `headless_renderer` example
## Bevy version
main
v0.14.0-dev
f91fd32

## Relevant system information

rustc/cargo 1.78.0

```ignore
`SystemInfo { os: "MacOS 14.4.1 ", kernel: "23.4.0", cpu: "Apple M2 Max", core_count: "12", memory: "96.0 GiB" }`
`AdapterInfo { name: "Apple M2 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }`
```

## What you did

Hello all - I am taking a look at the new headless_renderer example in the repo: https://github.com/bevyengine/bevy/blob/main/examples/app/headless_renderer.rs
https://github.com/bevyengine/bevy/pull/13006

I have a use case where I would need to set the dimensions of the output image based on something outside of bevy. So I ran the example after changing the configured dimensions (on line 69 in the AppConfig struct).

## What went wrong

I expected the output image to be properly formatted at the indicated resolution. However, just changing the width by 1 or a few results in output that is skewed/misaligned (examples attached).

I sort of suspected that this had to do with bevy's default output, which doesn't match the resolution in the example but does have the same aspect ratio. So I changed the primary_window in WindowPlugin from None to a window with the config-defined dimensions:

```rust
primary_window: Some(Window {
  resolution: WindowResolution::new(config.width as f32, config.height as f32),
  ..default()
}),
```

...but it didn't change the output at all (and in fact no window appears, I assume because the camera's target is still set to a texture). 

## Additional information

I assume its most likely some kind of bit alignment issue where the resolution needs to be a certain aspect ratio or along certain increments, but I tried changing the width by increments of 4, 8, etc, and had no luck.

Thank you.

### PR DESCRIPTION
**Pull request by bugsweeper**
# Objective

- Fixes #13384 .

## Solution

- If the image became wider when copying from the texture to the buffer, then the data is reduced to its original size when copying from the buffer to the image.

## Testing

- Ran example with 1919x1080 resolution

### CODE REVIEW

File: examples/app/headless_renderer.rs [hunk 1]
```diff
@@ -492,7 +491,20 @@ fn update(
                 for image in images_to_save.iter() {
                     // Fill correct data from channel to image
                     let img_bytes = images.get_mut(image.id()).unwrap();
-                    img_bytes.data.clone_from(&image_data);
+
+                    let row_bytes = img_bytes.width() as usize
```
Comment (alice-i-cecile) [inline examples/app/headless_renderer.rs:495]:
```suggestion
                    // We need to ensure that this works regardless of the image dimensions
                    let row_bytes = img_bytes.width() as usize
```

**Review by alice-i-cecile — COMMENTED** [review]
Fix seems fine, but I think it would be helpful to include the context about why we're doing this in the source itself.

**Review by mockersf — CHANGES_REQUESTED** [review]
It doesn't work.

If I ask for an image of 1080x1080:
```
thread '<unnamed>' panicked at examples/app/headless_renderer.rs:517:49:
range end index 4320 out of range for slice of length 2560
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Encountered a panic in system `headless_renderer::update`!
Encountered a panic in system `bevy_app::main_schedule::Main::run_main`!
```

File: examples/app/headless_renderer.rs [hunk 2]
```diff
@@ -492,7 +495,23 @@ fn update(
...
+                        * img_bytes.texture_descriptor.format.pixel_size();
+                    let aligned_row_bytes = RenderDevice::align_copy_bytes_per_row(row_bytes);
+                    if row_bytes == aligned_row_bytes {
+                        img_bytes.data.clone_from(&image_data);
+                    } else {
+                        // shrink data to original image size
+                        img_bytes.data = image_data
+                            .chunks(aligned_row_bytes)
+                            .flat_map(|row| &row[..row_bytes])
```
[the remark this answers is not in the document]
Comment (bugsweeper) [reply examples/app/headless_renderer.rs]: @mockersf Thank you for your suggestion!

**Comment by bugsweeper** [comment]
Looks like problem of 1080x1080 was in last row, which somehow is reminder of division by aligned_row_bytes. I cut off remaining row, this reinsurance works well alone, but i will keep them both.

**Comment by bugsweeper** [comment]
@mockersf May I ask you for review?

**Review by mockersf — APPROVED** [review]

**Pull request merged**